lib/ssl/doc/src/ssl.xml

```diff
@@ -356,13 +356,53 @@
      </datatype>
      
     <datatype>
-      <name name="key_password"/>
+      <name name="key_pem_password"/>
       <desc>
 	<p>String containing the user's password or a function returning same type. Only used if the
 	private keyfile is password-protected.</p>
       </desc>
     </datatype>
 
+    <datatype>
+      <name name="certs_keys"/>
+      <desc>
+	<p>A list of a certificate (or possible a certificate and its
+	chain) and the associated key of the certificate, that may be
+	used to authenticate the client or the server.  The
+	certificate key pair that is considered best and matches
+	negotiated parameters for the connection will be selected.
+	Different signature algorithms are prioritized in the order
+	<c> eedsa, ecdsa, rsa_pss_pss, rsa and dsa </c>.  If more than
+	one key is supplied for the same signing algorithm (which is
+	probably an unusual use case) they will prioritized by
+	strength unless it is a so called <c>engine key</c> that will
+	be favoured over other keys. As engine keys can not be
+	inspected suppling more than one engine key will make no
+	sense. This offers flexibility to for instance configure a
+	newer certificate that is expected to be used in most cases
+	and an older but acceptable certificate that will only be used
+	to communicate with legacy systems. Note that there is a trade
+	off between the induced overhead and the flexibility so
+	alternatives should be chosen for good reasons. If specified
+	overrides all single certificate and key options.
+	</p>
+
+	<note><p> <c>eddsa</c> certificates are only supported by TLS-1.3 that does not support  <c>dsa</c> certificates.
+	 <c>rsa_pss_pss</c> (RSA certificates using Probabilistic Signature Scheme) are supported in TLS-1.2 and TLS-1.3, but some
+	TLS-1.2 implementations may not support <c>rsa_pss_pss</c>.
+	</p></note>
+      </desc>
+    </datatype>
+
+    <datatype>
```

Comment:
There must be examples for how this is to be used and also how this option can replace the use of the old options for certificates and keys